Migrate to Android X.

JeffScaturro opened the pull request:
**Issue:** we need to migrate to Android X (#62).

**What was changed:** migrate to Android X ([reference guide](https://developer.android.com/jetpack/androidx/migrate)).

**Testing:** consume this change in a Android X project (example project has been migrated).

JeffScaturro commented:
Note that this will be a breaking change, and should require a major version bump. Since this plugin is not at 1.0.0 yet, a minor version bump acts as a major. We'd want this to go to `0.9.0`. I did not include `CHANGELOG.md` notes as I suspect you'd commit them with the release, however I can add them if you'd like.

FYA @hyochan

hyochan commented:
@JeffScaturro Thank you so much for the contribution! I'd like you to add the `changelogs` and also add [Migration Guide](https://github.com/dooboolab/react-native-iap#migration-guide) of your taste. I can edit this when it isn't suitable so please feel free to add whatever you like.

JeffScaturro commented:
@hyochan I added the changelog entry, listed the breaking change in the readme, and included a reference to the official migration guide in the Migration Guide section I added to the readme. I also bumped the pub version in this PR. Thanks!
